File: frontend/TideGuard/TideGuard/Model/FloodForecast.swift
//
//  FloodForecast.swift
//  TideGuard
//
//  Created by Faki Doosuur Doris on 12.03.2026.
//

import Foundation


//struct FloodForecast: Codable {
//
//    let date: String
//    let dayName: String
//    let tp: Double
//    let ro: Double
//    let t2m: Double
//    let swvl1: Double
//    let tp_7d: Double
//    let tp_14d: Double
//    let tp_30d: Double
//    let tp_7d_max: Double
//    let ro_7d: Double
//    let ro_14d: Double
//    let swvl1_3d_change: Double
//    let latitude: Double
//    let longitude: Double
//    let month: Int
//    let dayOfYear: Int
//}


struct FloodForecast: Codable {

    var date: String
    var dayName: String
    var tp: Double
    var ro: Double
    var t2m: Double
    var swvl1: Double
    var tp_7d: Double
    var tp_14d: Double
    var tp_30d: Double
    var tp_7d_max: Double
    var ro_7d: Double
    var ro_14d: Double
    var swvl1_3d_change: Double
    var latitude: Double
    var longitude: Double
    var month: Int
    var dayOfYear: Int
}

struct FloodRiskDay {
    let date: String
    let dayName: String
    let riskLevel: Int
    let probability: Double 
}

File: backend/TideGuard/src/main/java/com/example/tideguard/Models/FloodForecast.java
package com.example.tideguard.Models;


import lombok.Data;

@Data
public class FloodForecast {

    private String date;
    private String dayName;
    private double tp;
    private double ro;
    private double t2m;
    private double swvl1;
    private double tp_7d;
    private double tp_14d;
    private double tp_30d;
    private double tp_7d_max;
    private double ro_7d;
    private double ro_14d;
    private double swvl1_3d_change;
    private double latitude;
    private double longitude;
    private int month;
    private int dayOfYear;
    private double stateEncoded;

}
